### DISCUSSION TITLE
A few ideas here and there

### DISCUSSION THREAD

**Discussion by a0g83agbc84** [jeffvli/sonixd#262] [Ideas]
Hey!

First off, AWESOME subsonic client ;)

Second, list of ideas:
1. Option to make individual playlists offline (like in Spotify)
2. `Add to playlist` option show directly list of playlists instead of having to select it using the dropdown menu over and over again
3. Once a song has been added to a playlist (action finished) dismiss context menu
4. Port to Android! I think it's HTML5 so it should be fairly achievable, right? :P

Thanks again for this awesome client!

**Comment by jeffvli** [comment]
Thanks for the ideas!

1. Currently planned but no ETA as of yet (still thinking about how to implement)
2. Might be a good idea.. are you thinking of something like a vertically scrolling list of the playlists instead of the dropdown?
3. I think I kept it open in case the user wanted to add the song to multiple playlists (but maybe I can allow the user to select multiple playlists to add to)
4. Android port unfortunately not planned. I don't think it's an easy port unless as a web app, and even then there would need to be a lot of refactoring. I believe crossfade may also not be compatible with mobile devices due to how the volume is controlled on mobile.

**Comment by drtebi** [comment]
Hello,

I would like to chip in since I was looking for similar features.

In my opinion the easiest way to create playlists would be to have a split view (songs/albums left, playlist right), so that one can just browse the music library on the left and drag whatever s/he likes into the playlist on the right. That's more or less how it works in Cantata (an MPD client). This really makes creating playlists a breeze, since it only requires one click and one drag.

Partially this is already there—I can open the mini-view of my "now playing" playlist. However, I cannot drag albums or songs into it.... If that would be possible, the user could just create his "now playing" playlist and then choose all songs and save it as a new playlist.

Just some ideas. It might be worth checking out Cantata for inspiration, I think it's really well done there in my opinion.

**Reply by jeffvli** [reply]
@drtebi I agree, better playlist handling would be nice! 

My initial worry about this kind of functionality is that playlist saving on both Jellyfin and Subsonic are somewhat finnicky in the sense that the entire playlist needs to be deleted and re-saved whenever we want to re-order tracks. I'm thinking in order to implement it, the playlist would be be persisted locally in the client, and then a button would need to be added to save whenever you're done making changes (similar to how playlist save function works now).

Though in case you're talking about the now playing queue (as opposed to 'playlist'), that would probably be pretty easy to implement. 

I do want to add this functionality though, but it would would be coming later in the 1.0.0 version (which is still in the works!).

**Reply by drtebi** [reply]
Yes I was actually talking about the "now playing queue". Just adding a "save now playing queue as playlist" button would be great.

It would provide a very quick and easy way to create a playlist... one just browses his/her files, adds songs to the queue, and saves the queue as a playlist.

**Comment by isle9** [comment]
> 4. Port to Android! I think it's HTML5 so it should be fairly achievable, right? :P

If you need an Android app, try Symfonium, it's great.

**Reply by AdmnJ** [reply]
> Second, list of ideas:
> 
> 1. Option to make individual playlists offline (like in Spotify)

I would like to second number one, the download links are great but I would love for it to store and sort downloads automatically like the offline features do on other apps. The example that I use regularly on mobile would be "Finamp"

**Comment by LilMistix** [comment]
Hello !
It would be super cool to add the French translation (I can help if you want), and I would also like a fullscreen mode like spotify before. In full screen we just see the cover, the title, the album etc... (a bit like kodi)
Thank you for your attention :)
